x-pack/plugins/canvas/canvas_plugin_src/functions/common/switch.ts

```diff
@@ -43,12 +43,16 @@ export function switchFn(): ExpressionFunctionDefinition<
       },
     },
     fn(input, args) {
-      return from(args.case ?? []).pipe(
-        concatMap((item) => item()),
-        filter(({ matches }) => matches),
-        pluck('result'),
-        merge(defer(() => args.default?.() ?? of(input))),
-        take(1)
+      const cases = args.case?.map((item) => defer(() => item())) ?? [];
```

Comment:
According to the Expressions spec above, `case` is a required argument.  I'm curious why the Argument interface was changed in https://github.com/elastic/kibana/pull/100409 and if/why it needs to remain so.

@ppisljar I thought the types I had written in Canvas would through a TS error if `required` was true and the argument was optional... I may be misremembering, though.

Reply:
That's a good finding. I've corrected types because we had that check in the implementation and, for some reason, test cases for that. I remember @ppisljar asked me to correct all the optional types, but it seems like I forgot to correct this one here. 
Please see my changes in the latest commit.